lib/events.js

```diff
@@ -599,11 +599,17 @@ EventEmitter.prototype.eventNames = function eventNames() {
   return this._eventsCount > 0 ? ReflectOwnKeys(this._events) : [];
 };
 
-function arrayClone(arr, n) {
-  const copy = new Array(n);
-  for (let i = 0; i < n; ++i)
-    copy[i] = arr[i];
-  return copy;
+function arrayClone(arr) {
+  // At least since V8 8.3, this implementation is faster than the previous
+  // which always used a simple for-loop
+  switch (arr.length) {
+    case 2: return [arr[0], arr[1]];
```

Comment:
Hi,
It's just curiosity, why don't support the length 0 and 1 of the array in the switch-case ?

Reply:
Because those should never be the case when we have an array. Zero event handlers is represented by `undefined` and one event handler is stored by itself without an array.